src/main/java/seedu/address/ui/TaskCard.java

```diff
@@ -53,13 +54,38 @@ public TaskCard(Task task, int displayedIndex) {
         super(FXML);
         this.task = task;
         id.setText(displayedIndex + ". ");
+
         name.setText(task.getName());
+        LocalDateTime taskDeadline = task.getDateTime();
+        setTaskColor(taskDeadline);
+
         date.setText("Due: " + task.getDateTimeString());
 
         task.getTags().stream()
                 .sorted(Comparator.comparing(tag -> tag.tagName))
                 .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
 
+        setLink();
+    }
+
+    public void setTaskColor(LocalDateTime taskDateTime) {
+        LocalDateTime todayDate = LocalDateTime.now();
+        if (taskDateTime.isBefore(todayDate)) {
+            id.getStyleClass().add("cell_big_label_late");
+            name.getStyleClass().add("cell_big_label_late");
+            date.getStyleClass().add("cell_small_label_late");
+        } else if (taskDateTime.isBefore(todayDate.minusDays(3))) {
```

Comment:
Should this be taskDateTime.isAfter(todayDate.plusDay(3)) instead? Indicating that the task is reaching soon in 3 days?

Reply:
True